pkg/operator/ceph/controller/predicate.go

```diff
@@ -579,11 +579,14 @@ func WatchPredicateForNonCRDObject(owner runtime.Object, scheme *runtime.Scheme)
 				logger.Debugf("object %q matched on update", objectName)
 
 				// CONFIGMAP WHITELIST
-				// Only reconcile on rook-config-override CM changes
-				isCMTConfigOverride := isCMTConfigOverride(e.ObjectNew)
+				// Only reconcile on rook-config-override CM changes if the configmap is not empty
+				isCMTConfigOverride, configmapIsEmpty := isCMTConfigOverride(e.ObjectNew)
```

Comment:
I've been thinking about this. What if a user has a non-empty CM and then deletes the content? In that case, wouldn't we still want to reconcile to clear out what was previously set?

Reply:
I commented this on the issue as well, buthere is a little bit more to consider as well:

... this is a feature of Controller Runtime. It sets all resources to re-reconcile every 10 hours plus/minus a jitter factor. Reconciling periodically means that an operator can make sure that the current state of the system doesn't drift over time when there are no user changes to configs.

I don't think we have seen issues where users complain of drift over time with Rook, which is good. I think it's safe to make this change. But this may be an important note for the future if users ever start seeing system drift over time in otherwise stable clusters.

We

Reply:
> I've been thinking about this. What if a user has a non-empty CM and then deletes the content? In that case, wouldn't we still want to reconcile to clear out what was previously set?

In that case the workaround would be to restart the operator. This seems like enough of a corner case to justify the change if it helps the common case skip the unnecessary reconcile.

Reply:
Yeah, I don't think we want to restart the dashboard unnecessarily, and this is definitely an improvement to help that. It seemed like relevant info to note what is causing the reconcile.

I found a thread today where a controller-runtime main dev offered the opinion that the resync risks masking bugs, so not resyncing without user input is actually a good thing. 

> The problem was that nobody actually understood what this knob did -- it doesn't relist from the API server (which is what most people thought it did). All it does is force a re-reconcile of everything in the cache. Generally, you probably don't want to do this -- it's often expensive, it'll mask subtle bugs, and it's usually not necessary. It's actually probably fine to set this to never, except it might save you in production if you've got bugs in your reconcile logic because you didn't write a level-based controller.

(https://github.com/kubernetes-sigs/controller-runtime/issues/521#issuecomment-518457302)